### PR TITLE
chore/workspaces #66

### PR DESCRIPTION
**Pull request by jenbutongit**
# Description
closes: #66 
unblocks #84, #81, #82 

- Repo has now changed to `yarn 2`. You must now explicitly use yarn 2 for this repository.
- Changes to dev flow
  - please read readme. tldr, always run scripts in the root directory. I have "aliased" `yarn workspace @xgovformbuilder/runner` to `yarn runner` etc
- Changes to CI flow
  - There is now only one workflow that is triggered
  - CI has now consolidated the build, test and lint scripts. Persisting and attaching to workspace adds an additional 50s each time this happens
  - If model or engine changes, it will also build/test the downstream dependants 
  - If model/engine was unchanged, but their dependants were, rebuild them, but don't test. 
- version bump to 2.0.0
- changes to readme
- some tests have been disabled (repeat/n pages), I'll address them in my new feature branch

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [x] CI looks happy (bar linting issues)
- [x] Tests are running and passing
- [x] all running locally

# Checklist:

- [x] My code follows the [javascript standard style](https://standardjs.com/)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and versioning
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have rebased onto master and there are no code conflicts

### CODE REVIEW

File: designer/client/conditions/inline-conditions-definition.js [hunk 1]
```diff
@@ -129,26 +132,28 @@ class InlineConditionsDefinition extends React.Component {
...
+              className='govuk-select' id='cond-operator' name='cond-operator' value={condition.operator ?? ''}
+              onChange={this.onChangeOperator}
+            >
+              <option />
+              {
+                getOperatorNames(fieldDef.type).map(conditional => {
+                  return <option
+                    key={`${condition.field}-${conditional}`}
+                    value={conditional}
+                  >{conditional}
```
Comment (andymoody) [inline designer/client/conditions/inline-conditions-definition.js:145]: Yikes! Is this what eslint demands? It comes out pretty hard to read!

File: engine/yarn-error.log [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+Yarn version: 
+  1.22.4
+
+Node version: 
+  12.16.1
```
Comment (andymoody) [inline engine/yarn-error.log:11]: Do we want this file committed?

File: engine/test/.transform.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+const Babel = require('@babel/core')
```
Comment (andymoody) [inline engine/test/.transform.js:1]: One for interests sake; What is the purpose of these files, what do they do for us and why?

File: model/making-achange-to-model [hunk 4]
```diff
@@ -0,0 +1,2 @@
+another change
```
Comment (andymoody) [inline model/making-achange-to-model:1]: Was this just here for testing purposes and should now be removed?

File: runner/src/server/plugins/builder/pages/summary.js [hunk 5]
```diff
@@ -321,6 +321,10 @@ class SummaryViewModel {
     return result.value
   }
 
+  get webhookDataPaymentReference () {
```
Comment (andymoody) [inline runner/src/server/plugins/builder/pages/summary.js:324]: Was this part of some other changeset?

File: designer/client/conditions/inline-conditions-definition.js (hunk 1 above)
Comment (jenbutongit) [reply designer/client/conditions/inline-conditions-definition.js:145]: aye, we can always change the rules if it doesn't suit us. I'll have a look at what's causing it

File: runner/src/server/plugins/builder/pages/summary.js (hunk 5 above)
Comment (jenbutongit) [reply runner/src/server/plugins/builder/pages/summary.js:324]: eslint wants get/set pairs

File: engine/test/.transform.js (hunk 3 above)
Comment (jenbutongit) [reply engine/test/.transform.js:1]: I was having some trouble un-hoisting node_modules/@xgovformbuilder/lab-babel, so I've got the transform scripts in here. I'll refactor into model or a new shared /lib and remove @xgovformbuilder/lab-babel

File: engine/yarn-error.log (hunk 2 above)
Comment (jenbutongit) [reply engine/yarn-error.log:11]: nope. oops!

**Comment by andymoody** [comment]
Trying to run 'yarn runner dev' throws an error: 
```
Syntax Error: Expected "$", "$(", "${", "&&", "'", ";", "<", "<<<", ">", ">>", "\"", "\\", "|", "|&", "||", [ \t], or end of input but "&" found (line 1, column 22).
    at ie (/Users/andymoody/dev/home-office/digital-form-builder/.yarn/releases/yarn-berry.js:86:361659)
    at parse (/Users/andymoody/dev/home-office/digital-form-builder/.yarn/releases/yarn-berry.js:86:371853)
    at i (/Users/andymoody/dev/home-office/digital-form-builder/.yarn/releases/yarn-berry.js:2:413729)
    at F (/Users/andymoody/dev/home-office/digital-form-builder/.yarn/releases/yarn-berry.js:2:443423)
    at i.configuration.reduceHook.script (/Users/andymoody/dev/home-office/digital-form-builder/.yarn/releases/yarn-berry.js:2:339508)
    at /Users/andymoody/dev/home-office/digital-form-builder/.yarn/releases/yarn-berry.js:2:339630
    at async a.mktempPromise (/Users/andymoody/dev/home-office/digital-form-builder/.yarn/releases/yarn-berry.js:2:408207)
    at async Module.B (/Users/andymoody/dev/home-office/digital-form-builder/.yarn/releases/yarn-berry.js:2:339248)
    at async Se.execute (/Users/andymoody/dev/home-office/digital-form-builder/.yarn/releases/yarn-berry.js:2:78449)
    at async Se.validateAndExecute (/Users/andymoody/dev/home-office/digital-form-builder/.yarn/releases/yarn-berry.js:2:551219)
```

However I see it can now be started using 'yarn runner start'

File: .circleci/circle_trigger.sh [hunk 6]
```diff
@@ -107,6 +107,9 @@ fi
 
 echo "Changes detected in ${COUNT} package(s)."
 
+#PARAMETERS='"trigger":false, "designer":true, "runner":true, "engine":true'
```
[the remark this answers is not in the document]
Comment (jenbutongit) [reply .circleci/circle_trigger.sh:110]: this is commented out, I'll remove it though

**Review by andymoody — APPROVED** [review]

**Pull request merged**